atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/persistentlock/LockStoreImpl.java

```diff
@@ -82,7 +81,7 @@ public LockStore delegate() {
 
     private static final Logger log = LoggerFactory.getLogger(LockStoreImpl.class);
     private static final String BACKUP_LOCK_NAME = "BackupLock";
-    private static final AtomicBoolean isInitialized = new AtomicBoolean(false);
+    private volatile boolean isInitialized = false;
```

Comment:
Hmm. There seems to be a change in semantics here; if I create a transaction manager TM1, close it, and then create TM2 (all in the same JVM), then we previously fail but now we succeed (which makes sense). I guess that makes sense.

Reply:
Agreed that the current logic makes more sense. We should probably check that this won't break the internal large product though. Maybe running the tests with a local patch would be enough?